lang/php/lib/DataFile/AvroDataIOReader.php

```diff
@@ -74,16 +64,12 @@ class AvroDataIOReader
      *                             is not supported
      * @uses readHeader()
      */
-    public function __construct($io, $datum_reader)
-    {
-
-        if (!($io instanceof AvroIO)) {
-            throw new AvroDataIOException('io must be instance of AvroIO');
-        }
-
+    public function __construct(
+        AvroIO $io,
+        private AvroIODatumReader $datum_reader
```

Comment:
Why `AvroIO $io` is assigned manually with `$this->io = $io` but `$datum_reader` is not ?